rust_src/src/marker.rs

```diff
@@ -17,10 +17,33 @@ impl LispMarkerRef {
         // TODO: add assertions that marker_position in marker.c has.
         self.charpos
     }
+
+    pub fn charpos(self) -> Option<ptrdiff_t> {
```

Comment:
`charpos()` and `position()` methods are ambiguous. following the conventions, `charpos()` and `charpos_or_error()` would be better.